src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -706,17 +721,20 @@ public static SocketError GetSockOpt(SafeSocketHandle handle, SocketOptionLevel
                 return SocketError.Success;
             }
 
-            int optionLength = 4; // sizeof(int)
+            int optionLength = sizeof(int);
+            int tmpOptionValue;
             SocketError errorCode = Interop.Winsock.getsockopt(
                 handle,
                 optionLevel,
                 optionName,
-                out optionValue,
+                (byte*)&tmpOptionValue,
```

Comment:
Does `getsockopt` guarantee writing 0 to this on error? If not, this could disclose the contents of uninitialized stack memory in the error case.

Reply:
No, but the callers throw in the case of error. I can set it to 0 though if it would help you sleep better :)

Reply:
Derp. For some reason I was thinking that this was a public API. 😳

If the caller is Framework-owned code and never exposes the value outside of success cases then my care level drops considerably.